### PR TITLE
feat: Show dialog on full screen for first time

### PR DESCRIPTION
**Pull request by sonulohani**
## Description
This PR removes the ESC push button binding for exiting out of the full screen. Additionally, it will show the dialog for the user who has either accidentally pressed F11 or pressed F11 but doesn't know how to restore the window back to normal

## Related Issues / Pull Requests
https://github.com/cpeditor/cpeditor/issues/786

## Motivation and Context
Esc key will be used to cancel the selection

## How Has This Been Tested?
Ubuntu 21.04

## Screenshots (if appropriate)

image

## Checklist

- [x] If the key of a setting is changed, the `old` attribute is updated or it is resolved in SettingsUpdater.
- [x] If there are changes of the text displayed in the UI, they are wrapped in `tr()` or `QCoreApplication::translate()`.
- [ ] If needed, I have opened a pull request or an issue to update the [documentation](https://github.com/cpeditor/cpeditor.github.io).
- [x] If these changes are notable, they are documented in [CHANGELOG.md](https://github.com/cpeditor/cpeditor/blob/master/CHANGELOG.md).

### CODE REVIEW

File: src/Settings/settings.json [hunk 1]
```diff
@@ -1176,5 +1176,11 @@
     "type": "bool",
     "default": "false",
     "tip": "Automatically uncheck test cases when they get accepted."
+  },
+  {
+    "name": "On Full Screen Dialog Shown",
```
Comment (coder3101) [inline src/Settings/settings.json:1181]: Any issues with this name?
```suggestion
    "name": "Full Screen Dialog Shown",
```

File: src/appwindow.cpp [hunk 2]
```diff
@@ -1302,6 +1300,12 @@ void AppWindow::on_actionFullScreen_toggled(bool checked)
     auto state = windowState();
     state.setFlag(Qt::WindowFullScreen, checked);
     setWindowState(state);
+    if (!SettingsHelper::isOnFullScreenDialogShown())
+    {
+        QMessageBox::information(this, tr("How to exit full-screen"),
+                                 tr("Press the F11 key on your computer's keyboard to exit full-screen mode."));
```
Comment (coder3101) [inline src/appwindow.cpp:1306]:
```suggestion
                                 tr("Press the F11 key to exit full-screen mode."));
```

File: src/appwindow.cpp [hunk 3]
```diff
@@ -1302,6 +1300,12 @@ void AppWindow::on_actionFullScreen_toggled(bool checked)
     auto state = windowState();
     state.setFlag(Qt::WindowFullScreen, checked);
     setWindowState(state);
+    if (!SettingsHelper::isOnFullScreenDialogShown())
+    {
+        QMessageBox::information(this, tr("How to exit full-screen"),
```
Comment (coder3101) [inline src/appwindow.cpp:1305]:
```suggestion
        QMessageBox::information(this, tr("Hint: Exit Full screen mode"),
```

File: src/appwindow.cpp [hunk 4]
```diff
@@ -1302,6 +1300,12 @@ void AppWindow::on_actionFullScreen_toggled(bool checked)
...
+                                 tr("Press the F11 key on your computer's keyboard to exit full-screen mode."));
+        SettingsHelper::setOnFullScreenDialogShown(true);
```
Comment (coder3101) [inline src/appwindow.cpp:1307]: Should change as you change config file name.

File: src/appwindow.cpp (hunk 3 above)
Comment (ouuan) [reply src/appwindow.cpp:1305]: I prefer the current one.

Comment (coder3101) [reply src/appwindow.cpp:1305]: It's okay if you prefer. I am fine with either of them.

File: src/Settings/settings.json (hunk 1 above)
Comment (sonulohani) [reply src/Settings/settings.json]: Changed

File: src/appwindow.cpp (hunk 2 above)
Comment (sonulohani) [reply src/appwindow.cpp]: Changed

File: src/appwindow.cpp (hunk 4 above)
Comment (sonulohani) [reply src/appwindow.cpp]: Changed

**Comment by ouuan** [comment]
Please, use Qt Linguist to edit the translations, or remember to mark the blank translations as unfinished.

**Pull request merged**